easy-ui-react/src/Button/Button.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { ReactElement, ReactNode, useRef } from "react";
+import { IconProps } from "Icon";
+import { useButton } from "@react-aria/button";
+import { classNames, variationName } from "../utilities/css";
+
+import styles from "./Button.module.scss";
+
+export type ButtonColor =
+  | "primary"
+  | "secondary"
+  | "success"
+  | "warning"
+  | "neutral"
+  | "support"
+  | "inverse";
+export type ButtonAppearance = "filled" | "outlined" | "link";
+export type ButtonSize = "sm" | "md";
+
+export type ButtonProps = {
+  /** Button color */
+  color?: ButtonColor;
+  /** Button appearance */
+  appearance?: ButtonAppearance;
+  /** Button size */
+  size?: ButtonSize;
+  /** Disables button */
+  isDisabled?: boolean;
+  /** Button will grow to width of container */
+  isBlock?: boolean;
+  /** Positions icon before children */
+  startIcon?: ReactElement<IconProps>;
+  /** Positions icon after children */
+  endIcon?: ReactElement<IconProps>;
```

Comment:
i've seen `iconAtStart` and `iconAtEnd` as names recently in other libraries and i'm wondering if that's slightly better? 🤔